classes/sota_sanity.bbclass

```diff
@@ -10,6 +10,32 @@ def sota_check_required_variables(status, d):
         if not d.getVar(var):
             status.addresult("%s should be set in your local.conf.\n" % var)
 
+def sota_check_variables_validity(status, d):
+    var = d.getVar("OSTREE_BRANCHNAME")
+    if var != "":
```

Comment:
The non empty check is redundant.

More concise with regex: `if re.match("^[a-zA-Z0-9_-]*$", d.getVar("OSTREE_BRANCHNAME")) is not None: ...`

Reply:
thanks @lbonn , looks way better!